packages/mdc-textfield/mdc-textfield.scss

```diff
@@ -313,6 +317,95 @@ $mdc-textfield-disabled-border-on-dark: rgba(white, .3);
   // stylelint-enable plugin/selector-bem-pattern
 }
 
+.mdc-textfield--with-leading-icon {
+  .mdc-textfield__leading-icon:not([tabindex]),
+  .mdc-textfield__leading-icon[tabindex="-1"] {
+    pointer-events: none;
+    cursor: default;
+  }
+
+  .mdc-textfield__trailing-icon:not([tabindex]),
+  .mdc-textfield__trailing-icon[tabindex="-1"] {
+    pointer-events: none;
+    cursor: default;
+  }
+
+  .mdc-textfield__input {
+    @include mdc-rtl-reflexive-box(padding, left, $mdc-textfield-icon-padding);
+  }
+
+  .mdc-textfield__leading-icon {
+    position: absolute;
+    bottom: 16px;
+    cursor: pointer;
+
+    @include mdc-rtl-reflexive-position(left, $mdc-textfield-icon-position);
+
+    @include mdc-theme-dark(".mdc-textfield") {
+      @include mdc-theme-prop(color, text-secondary-on-dark);
+    }
+  }
+
+  .mdc-textfield__label {
+    @include mdc-rtl-reflexive-position(left, $mdc-textfield-icon-padding);
+  }
+}
+
+.mdc-textfield--with-trailing-icon {
+  .mdc-textfield__trailing-icon {
+    position: absolute;
+    bottom: 16px;
+    width: 24px;
```

Comment:
Why is width specified here but not for leading-icon? Is it needed in either case?

Also, should we combine these common styles under a single selector so it's not duplicated in both the Sass and resulting CSS? (I think the dark styling is also common)

```sass
  .mdc-textfield-with-leading-icon .mdc-textfield__leading-icon,
  .mdc-textfield-with-trailing-icon .mdc-textfield__trailing-icon {
    position: absolute;
    ...
  }
```

Reply:
We in fact, do not need it at all. Agreed on the selector consolidation.